src/overlays/actors/ovl_Obj_Spinyroll/z_obj_spinyroll.c

```diff
@@ -28,121 +40,661 @@ const ActorInit Obj_Spinyroll_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1EC38.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1ECC0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1ECD4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/ObjSpinyroll_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/ObjSpinyroll_Draw.s")
+s32 func_80A1E648(ObjSpinyroll* this, GlobalContext* globalCtx, s32 arg2) {
+    f32 sp1C;
+
+    if (func_80A1E3D8(this, globalCtx, &sp1C, arg2)) {
+        this->dyna.actor.world.pos.y += sp1C;
+        return true;
+    }
+    return false;
+}
+
+DynaPolyActor* func_80A1E694(ObjSpinyroll* this, GlobalContext* globalCtx) {
+    if (this->unk_3A4.unk_F0 != NULL) {
+        return DynaPoly_GetActor(&globalCtx->colCtx, this->unk_3A4.unk_F0->unk_00[0].bgId);
+    }
+    return NULL;
+}
+
+s32 func_80A1E6D4(ObjSpinyroll* this, GlobalContext* globalCtx) {
+    DynaPolyActor* temp_v0 = func_80A1E694(this, globalCtx);
+
+    if ((temp_v0 != NULL) && (Math3D_Vec3fDistSq(&temp_v0->actor.world.pos, &temp_v0->actor.prevPos) > 0.0001f)) {
+        return true;
+    }
+    return false;
+}
+
+void ObjSpinyroll_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjSpinyroll* this = THIS;
+    f32 sp44;
+    Path* path;
+    s32 pad2;
+    Vec3s* points;
+    Vec3s* sp34;
+    Vec3s* sp30;
+
+    sp44 = D_80A1F1B8[OBJSPINYROLL_GET_C000(&this->dyna.actor)];
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->dyna.actor.world.rot.x = 0;
+    this->dyna.actor.world.rot.z = 0;
+
+    this->dyna.actor.shape.rot.x = 0;
+    this->dyna.actor.shape.rot.z = 0;
+
+    this->dyna.actor.scale.x = 0.1f * sp44;
+    this->dyna.actor.scale.y = 0.1f;
+    this->dyna.actor.scale.z = 0.1f;
+
+    this->dyna.actor.uncullZoneScale = 250.0f * sp44;
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_spinyroll_Colheader_000E68);
+
+    Collider_InitTris(globalCtx, &this->collider);
+    Collider_SetTris(globalCtx, &this->collider, &this->dyna.actor, &sTrisInit, this->colliderElements);
+
+    if (OBJSPINYROLL_GET_7F(&this->dyna.actor) == OBJSPINYROLL_GET_7F_7F) {
+        func_80A1E9C4(this);
+        return;
+    }
+
+    this->dyna.actor.world.rot.y = 0;
+    func_80A1DFA0(this);
+    this->unk_4A4 = D_80A1F1C4[OBJSPINYROLL_GET_380(thisx)];
+
+    path = &globalCtx->setupPathList[OBJSPINYROLL_GET_7F(&this->dyna.actor)];
+    points = Lib_SegmentedToVirtual(path->points);
+
+    sp34 = &points[0];
+    sp30 = &points[1];
+
+    Math_Vec3s_ToVec3f(&this->unk_4AC[0], sp34);
+    Math_Vec3s_ToVec3f(&this->unk_4AC[1], sp30);
+    Math_Vec3f_Copy(&this->dyna.actor.world.pos, &this->unk_4AC[0]);
+
+    if ((sp34->x == sp30->x) && (sp34->y != sp30->y) && (sp34->z == sp30->z)) {
+        this->unk_49C = 1;
```

Comment:
```suggestion
        this->unk_49C = true;
```
This variable looks boolean, you can change the assignments and the `== 0` check